File: python/122_best_time_to_buy_and_sell_stock_ii.py
#!/usr/bin/env python3


class Solution:
    def maxProfit(self, prices):
        m = 0
        for i in range(1, len(prices)):
            m += prices[i] - prices[i-1] if prices[i] > prices[i-1] else 0
        return m


if __name__ == '__main__':
    print(Solution().maxProfit([7, 1, 5, 3, 6, 4]), 7)

File: c++/122_best_time_to_buy_and_sell_stock_ii.cpp
#include <gtest/gtest.h>
#include <iostream>
#include <vector>

class Solution {
   public:
    int maxProfit(const std::vector<int>& prices) {
        int sum = 0;
        for (int i = 1; i < prices.size(); i++) {
            if (prices[i] > prices[i - 1]) {
                sum += prices[i] - prices[i - 1];
            }
        }
        return sum;
    }
};

int main(int argc, char* argv[]) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

TEST(test, case1) {
    EXPECT_EQ(solution.maxProfit({7, 1, 5, 3, 6, 4}), 7);
}